### PR TITLE
add ipmi credential fields to zabbix_host module

### PR DESCRIPTION
**Pull request by JanKoppe**
##### SUMMARY

This module allows the definition of an IPMI interface, but there is no way to define the required credentials for such an interface.

This patch allows the configuration of IPMI credentials and privilege level for
zabbix hosts.

Because Zabbix defines these credentials on a per-host-basis, not per-interface-basis, these keys have not been tied to the interface section in this implementation.

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

modules/monitoring/zabbix_host

##### ANSIBLE VERSION

```
ansible 2.3.0.0
  config file = 
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Dec 21 2016, 07:16:46) [GCC 6.2.1 20160830]
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by JanKoppe** [comment]
Is there anything I can do to help get this along?

File: lib/ansible/modules/monitoring/zabbix_host.py
Comment (D3DeFi) [inline lib/ansible/modules/monitoring/zabbix_host.py]: Zabbix API doesn't document ipmi as a dictionary, rather as separate objects (`ipmi_authtype`, `ipmi_privilege`, `ipmi_username` and `ipmi_password`). I would like see them as a separate options to this module with a note that at least username and password must be used together.

Options `ipmi_authtype` and `ipmi_privilege` will need defaults, e.g. `ipmi_privilege=dict(type='int', required=False, default=2)` and `version_added:` too.

**Review by D3DeFi — CHANGES_REQUESTED** [review]
Tested against Zabbix server 3.2 & 3.4. Will need rebase due to merge conflicts (http://docs.ansible.com/ansible/latest/dev_guide/developing_rebasing.html)

**Comment by eikef** [comment]
Once rebased, this looks good to me. Whether or not the ipmi dict is changed to separate ipmi_*-parameters I have no real opinion on, either looks fine to me.

**Comment by JanKoppe** [comment]
Hello,

I will try to incorporate @D3DeFi 's suggestions in the coming days and rebase this PR. I'm a bit busy right now.

Thanks a lot for your reviews so far!

**Comment by eikef** [comment]
@JanKoppe: if you want I can try to rebase it as well if you are short on time. Please let me know if you prefer that :)

**Comment by JanKoppe** [comment]
@eikef: I'm working on it now. After thinking about the keys, I'd say I would prefer having the configuration nested in the ipmi dict. This makes loading the credentials from a var a bit easier IMO.

**Comment by JanKoppe** [comment]
Should be good now. Let me now if you need any changes.

**Comment by eikef** [comment]
!needs_revision

**Comment by eikef** [comment]
shipit

**Comment by eikef** [comment]
bot_status

**Comment by eikef** [comment]
@D3DeFi this is currently blocked on your requested changes :-)

**Comment by D3DeFi** [comment]
@JanKoppe @eikef does it work for you when updating existing host? I was able to create new host with ipmi credentials, but when I create the host first and then try to update it I just get OK from ansible and nothing changes. 

Same goes for host, which already has some username/pass configured in IPMI tab and I want to change it via this module.  

Tested against Zbx server 3.2

**Comment by eikef** [comment]
@D3DeFi That is indeed correct, and it is not the only option this is true for. Basically there is a check missing in check_all_properties.

My intention is to go over that part of the code again and fix those problems in any case.

**Comment by D3DeFi** [comment]
@JanKoppe any update on the requested changes? We've just received a new PR introducing same functionality, but not lacking idempotency and ability to update/wipe IPMI settings.

**Comment by eikef** [comment]
Closing in favor of #33435. Thank you in any case. Sorry it did not get in earlier.

close_me

**Pull request closed without merging**
